src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +251,230 @@ def inner(tree, path, root_path):
+    misspelled = spell.unknown([word])
+    if misspelled:
+        for name in misspelled:
+            correct_spell.append(spell.correction(name))
+            possible_corrections.extend(list(spell.candidates(name)))
+        if match_whole_word and correct_spell == possible_corrections:
+            return correct_spell
+        elif match_case:
+            corrections_in_tree = set()
+            for correction in possible_corrections:
+                for name in names:
+                    if correction in name:
+                        corrections_in_tree.add(correction)
+            return list(corrections_in_tree)
+    else:
+        return [word]
+
+
+def search(
+    search_string: str,
+    language: Optional[str] = "eng",
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = True,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueConflict
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueConflict()
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off semantic and wildcard matching.",
+            UserWarning,
+        )
```

Comment:
For this `elif` to match, `language` must be falsy, since otherwise the previous one would have matched. IOW, the user has not chosen semantic mention so we should avoid mentioning it.

Reply:
Right, we have updated it as "``match_whole_word=True`` will turn off wildcard matching." Thank you.